docs/content/developer/standards/closed-loop-token/tutorial.mdx

```diff
@@ -0,0 +1,167 @@
+passing the rule witness, the token policy and its capability, and a new [bag](../../../references/framework/iota-framework/bag.mdx) in which we want to store the addresses.
+
+Then we can get the mutable config with our helpter function config_mut which is just a wrapper of `token::rule_config_mut`.
+And as a last step we add the address to our bag.
+
+:::note Bonus Task
+
+You could also implement additional functions to remove addresses from the rule configuration.
+
+:::
+
+Next we have to create functionality to stamp the action request if the rule is met. This is done by adding a `verify` function to the rules module:
+
+```move file=<rootDir>/docs/examples/move/clt-tutorial/sources/rules.move#L26-L56
+```
+
+Let's break it down:
+
+1) We do some basic checks and task like checking if the policy has a rule config. We get the config, sender and receiver.
+2) We split the verification into two parts: 
+    - If the action is a spend_action someone is trying to return the token to the municipality. We check if the sender is in our list. If this is true, we stamp the action request. If not we abort as we only want to allow shops to return the token.
+    - If the action is a transfer_action someone is trying to buy a LED bulb, so we check if the sender is in our list. If this is true, that means a shop is trying to spend our token. We can't allow that, so we apport and therefore never stamp the action request. We also check if the receiver is a shop.
+
+3) If we don't abort, we stamp the action request.
+
+## Back to our voucher module
+
+Now that we created the rules modules let's use it in our voucher module.
+We can import the rules like this:
+
+```move file=<rootDir>/docs/examples/move/clt-tutorial/sources/clt_tutorial.move#L6
+```
+
+Now we need to register the rule for the needed actions in our `init` function. Just add the following lines between the policy creation and the sharing:
+
+```move file=<rootDir>/docs/examples/move/clt-tutorial/sources/clt_tutorial.move#L37-L39
+```
+
+We are defining our rule for the default `spend` and `transfer` actions here. 
+But we could also pass any other action as a string here and make the rule work for custom functions of our module.
+
+The municipality also needs a way to register shops so we add a function for that which internally calls the `add_addresses` function from our rules module:
+
+```move file=<rootDir>/docs/examples/move/clt-tutorial/sources/clt_tutorial.move#L48-L55
+```
+
+And now we are back to our `buy_ledbulb` function. We can now verify/stamp the action request with the rules capability:
+
+```move file=<rootDir>/docs/examples/move/clt-tutorial/sources/clt_tutorial.move#L90
+```
+
+In this tutorial we are returning the action request. We could also approve it right away. 
+So in this case in a PTB we have to call the `token_policy::confirm_request` function afterwards which will check the request for the approval stamp and make our TX succeed.
```

Comment:
```suggestion
So, in this case, we have to call the `token_policy::confirm_request` function in a PTB afterward, which will check the request for the approval stamp and make our TX succeed.
```
LInk to /developer/iota-101/transactions/ptb/programmable-transaction-blocks-overview